crates/rpc/rpc-types/src/eth/error.rs

```diff
@@ -21,3 +27,51 @@ impl EthRpcErrorCode {
         }
     }
 }
+
+#[derive(Debug, thiserror::Error, Clone, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub enum RevmError {
```

Comment:
This should rly be using the executor error enum instead of writing its own: https://github.com/paradigmxyz/reth/blob/d774d914ceb76ee7d896f348f20e20d1ece5d92f/crates/interfaces/src/db.rs#L3

Reply:
> Ideally the RPC would use an ExecutorFactory like the rest of the code

RPC does not write to Disk and needs to configure the env directly depending on the RPC call.

The EvmEnvProvider trait is used by RPC to get block and cfg of the env and `create_txn_env` configures the txenv based on the request.